notebook/services/contents/fileio.py

```diff
@@ -104,7 +114,7 @@ def atomic_writing(path, text=True, encoding='utf-8', log=None, **kwargs):
         if os.name == 'nt' and os.path.exists(path):
             # Rename over existing file doesn't work on Windows
             os.remove(path)
```

Comment:
This bit should be moved into `replace_file` - the idea is that `os.replace()` can atomically replace `path` if it already exists, whereas `os.rename()` can't on Windows. So we only need this extra check when `os.replace()` isn't available.